src/Agent.Worker/Build/TfsVCSourceProvider.cs

```diff
@@ -239,7 +240,9 @@ public sealed class TfsVCSourceProvider : SourceProvider, ISourceProvider
                 {
                     // Create the comment file for reshelve.
                     StringBuilder comment = new StringBuilder(tfShelveset.Comment ?? string.Empty);
-                    if (!(executionContext.Variables.Build_GatedRunCI ?? true))
+                    string runCi = GetEndpointData(endpoint, Constants.EndpointData.GatedRunCI);
+                    bool? gatedRunCi = executionContext.Variables.GetBoolean(runCi);
```

Comment:
You're wrong :). GetBoolean [calls Get(name)](https://github.com/Microsoft/vsts-agent/blob/master/src/Agent.Worker/Variables.cs#L190).

StringUtil.ConvertToBoolean is the helper method you are looking for:
```charp
bool gatedRunCI = StringUtil.ConvertToBoolean(
    value: GetEndpointData(endpoint, Constants.EndpointData.GatedRunCI),
    defaultValue: true);
if (!gatedRunCI)
```